### PR TITLE
Updated question model

### PR DESCRIPTION
**Pull request by Deepankar-promact**
**Fixed Issues**
#21219

**Files Changed**
SingleMultipleAnswerQuestionApplicationClass.cs - changed base class to Question.cs
CodeSnippetQuestion.cs - changed base class to BaseModel.cs and added foreign key from Question.cs
Question.cs - Renamed from QuestionBase.cs to Question.cs
SingleMultipleAnswerQuestion.cs - Added foreign key from Question.cs
QuestionRepository.cs - Added To-Do comment 

**Checks**
- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [ ] Client side code comments (proper English, grammar and no spelling mistakes)
- [ ] Unused variables, methods, blank spaces and name spaces.
- [ ] Optimal Code and code formatting
- [ ] Commit History is proper, no merge is done.
- [ ] All test cases are executed provided by tester.

### CODE REVIEW

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
         public string QuestionDetail { get; set; }
+
         [Required]
         public QuestionType QuestionType { get; set; }
+
         [Required]
         public DifficultyLevel DifficultyLevel { get; set; }
+
         [Required]
         public string CreateBy { get; set; }
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs]: please remove created by and updated by.

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
+
         public string UpdatedBy { get; set; }
-        [Required]
+
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs]: why have you removed required validator?

Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs]: ForiegnKey doesn't require [Required] annotation

Comment (Deepankar-promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs]: ForeignKey doesn't require [Required] annotation

File: Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
-        [Required]
-        public string CreateBy { get; set; }
-        public string UpdatedBy { get; set; }
-        [Required]
+        
         public int CategoryID { get; set; }
+
+        [ForeignKey("CategoryID")]
         public virtual Category.Category Category { get; set; }
+
```
Comment (Krunal-Promact) [inline Trappist/src/Promact.Trappist.DomainModel/Models/Question/Question.cs]: please remove unnecessary space

**Pull request merged**